Mensaje de error inentendible cuando faltan tipar los parámetros

Cuando no se escribe el tipo de algún parámetro se obtiene el error:

`
/tmp/tmp.KxLAN4aE4Y/SubmissionTest.java:22: error: expected String responderA(persona) { ^ 1 error !!TEST FINISHED WITH COMPILATION ERROR!!`

--- a/lib/feedback_hook.rb
+++ b/lib/feedback_hook.rb
@@ -20,6 +20,12 @@ class JavaFeedbackHook < Mumukit::Hook
       missing_character result, '\{'
     end
 
+    def explain_missing_parameter_type(_, result)
+      (/#{error} <identifier> expected#{near_regex}/.match result).try do |it|
+        {near: it[1]}
+      end
+    end
+
     def explain_missing_return_statement(_, result)
       (/#{error} missing return statement#{near_regex}/.match result).try do |it|
         {near: it[1]}

--- a/lib/locales/en.yml
+++ b/lib/locales/en.yml
@@ -2,6 +2,7 @@ en:
   missing_semicolon: "Missing ';' near `%{near}`"
   missing_parenthesis: "Missing '(' near `%{near}`"
   missing_bracket: "Missing { near `%{near}`. Maybe you misspelled a class or method declaration."
+  missing_parameter_type: "Missing parameter type near `%{near}`"
   cannot_find_symbol: "Cannot find `%{symbol}` declaration in `%{location}`"
   missing_return_statement: "There is a method that must return something, but doesn't. Check your code again!"
   incompatible_types: "Class `%{down}` should be a `%{up}`. Probably an _extends_ or _implements_ is missing near `%{near}`."

--- a/lib/locales/es.yml
+++ b/lib/locales/es.yml
@@ -2,6 +2,7 @@ es:
   missing_semicolon: "Parece que falta un ';' cerca de `%{near}`"
   missing_parenthesis: "Parece que falta un '(' cerca de `%{near}`"
   missing_bracket: "Se esperaba una { cerca de `%{near}`. Fijate si tal vez, introdujiste un paréntesis de más o está mal escrita la declaración de clase o método."
+  missing_parameter_type: "Parece que falta el tipo de un parámetro cerca de `%{near}`"
   cannot_find_symbol: "No se encontró la definición de `%{symbol}` en `%{location}`"
   missing_return_statement: "Hay un método que debería retornar algo, pero no está retornando nada. ¡Revisá bien tu código!"
   incompatible_types: "La clase `%{down}` debería ser un `%{up}`. Revisá si no te falta un _extends_ o _implements_ cerca de `%{near}`."

--- a/spec/feedback_runner_spec.rb
+++ b/spec/feedback_runner_spec.rb
@@ -85,4 +85,18 @@ describe JavaFeedbackHook do
 
     it {expect(feedback).to include("* La clase `Foo` debería ser un `Bar`. Revisá si no te falta un _extends_ o _implements_ cerca de `Bar bar = new Foo();`.")}
   end
+  context 'missing parameter type' do
+    let(:request) {req(%q{
+    class Foo {
+      public int plusTwo(aNumber) {
+        return 2 + aNumber;
+      }
+    }}, %q{
+      public void testFoo(){
+        Assert.assertEquals(2, new Foo().plusTwo(3));
+      }
+    })}
+
+    it {expect(feedback).to include("* Parece que falta el tipo de un parámetro cerca de `public int plusTwo(aNumber) {`")}
+  end
 end

